app/abilities/ability.rb

```diff
@@ -43,7 +43,7 @@ def initialize(user)
         can :update, DiaryEntry, :user => user
         can [:create], DiaryComment
         can [:show, :create, :destroy], Follow
-        can [:read, :create, :mark, :unmute, :destroy], Message
+        can :manage, Message
```

Comment:
This shouldn't be removing `unmute` in the first commit as it doesn't move to it's own controller until the third commit.

Reply:
And it didn't because `:manage` grants all abilities.

Rewrote it to not use `:manage`. Marks and mutes now authorize messages as parent resources, which means they check `:show` (included in `:read`) on messages.